ui/jstests/test-learning-resource.jsx

```diff
@@ -158,28 +164,46 @@ define(['QUnit', 'jquery', 'learning_resources', 'reactaddons',
 
           // two terms, first vocab
           var $terms1Select = $($vocabSelect[0]).find("option");
-          assert.equal($terms1Select.size(), 3);
-          assert.equal($terms1Select[0].selected, true);
-          assert.equal($terms1Select[1].selected, false);
-          assert.equal($terms1Select[2].selected, false);
+          assert.equal($terms1Select.size(), 2);
+          assert.equal($terms1Select[0].selected, false);
+          assert.equal($terms1Select[0].selected, false);
 
           // TestUtils.Simulate.change only simulates a change event,
           // we need to update the value first ourselves
-          $vocabSelect[0].value = "hard";
+          $($vocabSelect[0]).val("hard").trigger('change');
           React.addons.TestUtils.Simulate.change($vocabSelect[0]);
           component.forceUpdate(function() {
             assert.equal($terms1Select[0].selected, false);
-            assert.equal($terms1Select[1].selected, false);
-            assert.equal($terms1Select[2].selected, true);
-
-            // make sure second vocab (which has a default value) is set properly
-            var $terms2Select = $($vocabSelect[1]).find("option");
-            assert.equal($terms2Select.size(), 2);
-            assert.equal($terms2Select[0].selected, false);
+            assert.equal($terms1Select[1].selected, true);
+          });
+          // make sure second vocab (which has a default value) is set properly
+          var $terms2Select = $($vocabSelect[1]).find("option");
+          assert.equal($terms2Select.size(), 2);
+          assert.equal($terms2Select[0].selected, true);
+          assert.equal($terms2Select[1].selected, false);
+          //the second vocabulary can be a multi select
+          $($vocabSelect[1]).val(["notrequired", "required"]).trigger('change');
+          React.addons.TestUtils.Simulate.change($vocabSelect[1]);
+          component.forceUpdate(function() {
+            assert.equal($terms2Select[0].selected, true);
```

Comment:
`done` must be the very last thing called in the test. Otherwise the test completes without executing the callbacks that are queued. Can you move that within the callback too?